### PR TITLE
[packages] Move all SofaComponent* + rename SofaAllCommonComponents

### PR DESCRIPTION
**Pull request by guparan**
This is an old work that has already been merged into ng branch (see #620) but since the NG task is paused for now I think it would be appreciable to have this in master.

The idea is to remove SofaComponent* from modules (because they are not modules) and put them next to SofaAllCommonComponents.

Concretely, this PR does:
- Move SofaComponent* next to SofaAllCommonComponents
- Rename SofaAllCommonComponents into SofaComponentAll
- Rename "packages" into "collections" to avoid confusion with CMake packages like SofaGeneral, SofaFramework, ...
- Somes fixes in SofaPreconditioner and SofaSparseSolver completing #1137 

______________________________________________________

This PR: 
- [ ] builds with SUCCESS for all platforms on the CI.
- [ ] does not generate new warnings.
- [ ] does not generate new unit test failures.
- [ ] does not generate new scene test failures.
- [ ] does not break API compatibility.
- [ ] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

### CODE REVIEW

File: SofaKernel/framework/sofa/helper/system/gl.h [hunk 1]
```diff
@@ -22,10 +22,10 @@
 #ifndef SOFA_HELPER_SYSTEM_GL_H
 #define SOFA_HELPER_SYSTEM_GL_H
 
+#include <sofa/helper/system/config.h>
```
Comment (guparan) [inline SofaKernel/framework/sofa/helper/system/gl.h:25]: Sorry, this is a fix that has nothing to do with my PR.
Still, it is needed when SOFA_NO_OPENGL is ON.

**Comment by guparan** [comment]
[ci-build][with-all-tests]

File: SofaKernel/SofaFramework/SofaMacros.cmake [hunk 2]
```diff
@@ -567,7 +570,7 @@ macro(sofa_install_targets package_name the_targets include_install_dir)
         get_target_property(target_sources ${target} SOURCES)
         #list(FILTER ${target_sources} INCLUDE REGEX ".*\.h\.in$") # CMake >= 3.6
         foreach(filepath ${target_sources})
-            if(${filepath} MATCHES ".*\.h\.in$")
+            if(filepath AND ${filepath} MATCHES ".*.h.in$")
```
Comment (damienmarchal) [inline SofaKernel/SofaFramework/SofaMacros.cmake]: No no...this is not the correct fix. See #1161  for the proper one.

Comment (guparan) [reply SofaKernel/SofaFramework/SofaMacros.cmake]: Ok thanks, I'll revert this :+1:

**Pull request merged**